test/test-platform-output.c

```diff
@@ -197,5 +198,12 @@ TEST_IMPL(platform_output) {
   printf("  version: %s\n", uname.version);
   printf("  machine: %s\n", uname.machine);
 
+#if defined(__APPLE__)
+  err = uv_battery_info(&battery);
+  printf("uv_battery_info:\n");
+  printf("  charging: %d\n", battery.charging);
+  ASSERT_OK(err);
+#endif
```

Comment:
```suggestion
  err = uv_battery_info(&battery);
  printf("uv_battery_info:\n");
  printf("  charging: %d\n", battery.charging);
  ASSERT_OK(err);
```
And add a stub for other platforms. The stub should probably return UV_ENOSYS, to indicate it's unimplemented.

Reply:
WIll do!

Reply:
Will change the assertion to this, in the meantime `ASSERT_OK(err && err == UV_ENOSYS);`

Reply:
I moved this file into the `unix` folder and renamed it just `battery.c` Using macros we can populate the struct, so we just handle two files, one for UNIX, and another one for Win.